scripts/cmake/kconfig.cmake

```diff
@@ -1,9 +1,70 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
-# Kconfig targets
-
 include(${CMAKE_CURRENT_LIST_DIR}/defconfigs.cmake)
 
+
+           ### configure-time .config ###
+
+
+if(NOT INIT_CONFIG_found)
+set(INIT_CONFIG "initial.config" CACHE STRING "Initial .config file")
+
+find_file(INIT_CONFIG_found
+	NAMES ${INIT_CONFIG}
+	NO_CMAKE_FIND_ROOT_PATH
+	NO_DEFAULT_PATH
+	PATHS "." "src/arch/xtensa/configs"
+)
+endif()
+
+if(NOT INIT_CONFIG_found)
+message(FATAL_ERROR
+	"Initial configuration missing, no ${INIT_CONFIG} found. "
+	"Provide a ${PROJECT_SOURCE_DIR}/initial.config file or specify some "
```

Comment:
```suggestion
	"Provide a ${PROJECT_SOURCE_DIR}/initial.config file without -DINIT_CONFIG or specify some "
```

Reply:
It is possible to override `${PROJECT_SOURCE_DIR}/initial.config` with `-DINIT_CONFIG`, as usual the command line argument has precedence over the default value.